### ISSUE TITLE
Cleanup WIN_CreateCursor() code a bit

### ISSUE THREAD

**Issue by DJm00n** [libsdl-org/SDL#6476]
https://github.com/libsdl-org/SDL/blob/92b3c53c92971e685254fd89f89ce6bde8cea60e/src/video/windows/SDL_windowsmouse.c#L94-L96

https://github.com/libsdl-org/SDL/blob/92b3c53c92971e685254fd89f89ce6bde8cea60e/src/video/windows/SDL_windowsmouse.c#L120

Not an issue but could be written like this:
```cpp
// hbmMask is a monochrome DIB (1 bit per pixel).
// Proper size calculation is tricky since its every
// scanline should be DWORD-aligned.
// Gery, Ron. “DIBs and Their Use.” MSDN Technology Group, 20 Mar. 1992, https://learn.microsoft.com/previous-versions/ms969901(v=msdn.10). 
UINT maskbitslen = (((surface->w + 31) & ~31) >> 3) * surface->h;
```

**Comment by slouken** [comment]
That's actually a functional change on systems with 64-bit size_t. Have you checked to make sure that's correct?

**Comment by DJm00n** [comment]
hm. I'll do some tests.

Seems https://devblogs.microsoft.com/oldnewthing/20101021-00/?p=12483 says that we should generate `hbmMask` properly, not only as 0xff data etc.

Here is proper code how this alpha channel bitmask should be generated:
https://github.com/dotnet/wpf/blob/89d172db0b7a192de720c6cfba5e28a1e7d46123/src/Microsoft.DotNet.Wpf/src/PresentationFramework/MS/Internal/AppModel/IconHelper.cs#L283-L347

Also [CreateBitmap ](https://learn.microsoft.com/en-us/windows/win32/api/wingdi/nf-wingdi-createbitmap) docs says that `Each scan line in the rectangle must be word aligned (scan lines that are not word aligned must be padded with zeros)`.

**Comment by DJm00n** [comment]
After a bit of research seems CreateBitmap docs are wrong. Created PR to fix that. Waiting for approve from MS: https://github.com/MicrosoftDocs/sdk-api/pull/1357

**Comment by slouken** [comment]
@DJm00n, I had to revert your change because it caused the default blank cursor to be black. Can you take a look?

Also, it looks like your change also removed the code that handled cursor scaling:
```c
    /* The cursor returned by CreateIconIndirect does not respect system cursor size
        preference, use CopyImage to duplicate the cursor with desired sizes */
    hcursor = CopyImage(hicon, IMAGE_CURSOR, surface->w, surface->h, 0);
```

**Comment by DJm00n** [comment]
@slouken `CopyImage` call is useless because it was added in 301819cd1d7dac2d0e953a6f1780d234431d9f75 and then changed current that does nothing in 1085c317a5088b5aba3245b2d4d59ca1cdcf8887.

I added another pass on this task. `SDL_CalculateShapeBitmap` is not fit because it calculates mask that is not expected by Win32.

**Comment by slouken** [comment]
It doesn't do nothing, it scales the cursor based on system settings. Check out https://github.com/libsdl-org/SDL/issues/5198

**Comment by DJm00n** [comment]
@slouken  `CreateIconIndirect` creates cursor of provided size and don't apply any DPI settings.
In 301819cd1d7dac2d0e953a6f1780d234431d9f75 (PR #4927 and was merged to 2.0.18) added call `CopyImage` with `LR_DEFAULTSIZE` - which scales created cursor according to the system metrics (DPI setting applied).

#5198 bug report was created that claims this DPI setting change in 2.0.18 as inappropriate. Fixed in 1085c317a5088b5aba3245b2d4d59ca1cdcf8887 by removing `LR_DEFAULTSIZE`. After this change  whole call to `CopyImage` is useless (and comment near it is misleading).

**Comment by slouken** [comment]
Ah, okay, thanks for clarifying. We should match the default Windows behavior for cursor sizes, but we can come back to that later.
